![image info](https://user-images.githubusercontent.com/91945811/146929832-e78ff1f4-2739-41b1-acd3-019a39b6a42c.png)


## 🐧 Statistik-Grundlagen mit dem `penguins`-Datensatz  

Dieser Datensatz stammt von realen Forschungen auf den Palmer-Inseln in der Antarktis. Wissenschaftler*innen haben dort **verschiedene Pinguinarten vermessen**, um biologische Unterschiede zu verstehen – z. B. um Arten voneinander zu unterscheiden, Geschlechter zu erkennen oder den Gesundheitszustand zu beurteilen.

---

## 💡 Was können wir damit lernen?

| Statistik-Methode         | Beispiel-Frage aus dem echten Leben |
|---------------------------|-------------------------------------|
| **Durchschnittswerte** (Mean, Median) | Welche Art ist im Schnitt schwerer – Adelie, Chinstrap oder Gentoo?  
| **Standardabweichung** | Welche Art zeigt mehr Variation beim Gewicht?  
| **Gruppenvergleiche** | Gibt es Größenunterschiede zwischen Männchen und Weibchen?  
| **Fehlende Werte** | Warum fehlen manchmal Werte? Hat das vielleicht mit Art oder Insel zu tun?  
| **Ausreißer** | Ist ein besonders schwerer Pinguin einfach ein Tippfehler – oder vielleicht ein Alpha-Tier?  
| **Korrelationen** | Gibt es einen Zusammenhang zwischen Flipperlänge und Körpermasse? Kann man aus Maßen auf das Gewicht schließen?

---

## 🧠 Lernziel – Was du nach dem Tutorial besser kannst:

- Du verstehst, **wie du einen Datensatz analysierst**, bevor du statistische Schlüsse ziehst.
- Du erkennst **Verteilungen, Mittelwerte, Streuungen, Zusammenhänge**.
- Du lernst, **wie fehlende oder extreme Werte** deine Analyse beeinflussen können.
- Du bekommst ein Gefühl dafür, **wie Statistik im echten Leben** eingesetzt wird – z. B. in Biologie, Medizin oder Data Science.

---


„Stell dir vor, du arbeitest in einem Naturschutzprojekt. Du hast über 300 Pinguine vermessen – aber die Daten sind unvollständig. Bevor du deine Ergebnisse veröffentlichst, willst du sicherstellen, dass du die **Datenlage wirklich verstehst**: Welche Arten sind am stärksten, welche Tiere sind auffällig, und was kann ich aus diesen Zahlen bereits lernen, ohne sie zu visualisieren?“

---

## 📦 1. Daten laden & Überblick gewinnen

In [4]:
import pandas as pd
import seaborn as sns

# Datensatz laden
df = sns.load_dataset("penguins")

# Ersten Eindruck verschaffen
df.head()

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,Male
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,Female
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,Female
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,Female


In [5]:
df.sample(10)

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
17,Adelie,Torgersen,42.5,20.7,197.0,4500.0,Male
146,Adelie,Dream,39.2,18.6,190.0,4250.0,Male
85,Adelie,Dream,41.3,20.3,194.0,3550.0,Male
76,Adelie,Torgersen,40.9,16.8,191.0,3700.0,Female
72,Adelie,Torgersen,39.6,17.2,196.0,3550.0,Female
113,Adelie,Biscoe,42.2,19.5,197.0,4275.0,Male
284,Gentoo,Biscoe,45.8,14.2,219.0,4700.0,Female
246,Gentoo,Biscoe,44.5,14.3,216.0,4100.0,NaN
228,Gentoo,Biscoe,43.3,13.4,209.0,4400.0,Female
263,Gentoo,Biscoe,49.6,15.0,216.0,4750.0,Male


In [6]:
display(df)

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,Male
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,Female
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,Female
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,Female
...,...,...,...,...,...,...,...
339,Gentoo,Biscoe,NaN,NaN,NaN,NaN,NaN
340,Gentoo,Biscoe,46.8,14.3,215.0,4850.0,Female
341,Gentoo,Biscoe,50.4,15.7,222.0,5750.0,Male
342,Gentoo,Biscoe,45.2,14.8,212.0,5200.0,Female


---

## ❓ Forschungsfrage 1: **Wie vollständig und sauber sind unsere Daten?**

### ▶️ Gibt es fehlende Werte?

In [8]:
df.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

🔍 **Antwort:**  
Ja, es fehlen Werte in den Spalten `sex`, `bill_length_mm`, `bill_depth_mm`, `flipper_length_mm` und `body_mass_g`.

---

### ▶️ Zeige alle Zeilen mit fehlenden Werten

In [10]:
df[df.isnull().any(axis=1)]

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
8,Adelie,Torgersen,34.1,18.1,193.0,3475.0,NaN
9,Adelie,Torgersen,42.0,20.2,190.0,4250.0,NaN
10,Adelie,Torgersen,37.8,17.1,186.0,3300.0,NaN
11,Adelie,Torgersen,37.8,17.3,180.0,3700.0,NaN
47,Adelie,Dream,37.5,18.9,179.0,2975.0,NaN
246,Gentoo,Biscoe,44.5,14.3,216.0,4100.0,NaN
286,Gentoo,Biscoe,46.2,14.4,214.0,4650.0,NaN
324,Gentoo,Biscoe,47.3,13.8,216.0,4725.0,NaN
336,Gentoo,Biscoe,44.5,15.7,217.0,4875.0,NaN


In [11]:
df["sex"] = df["sex"].fillna("Unknown")

In [12]:
df[df.isnull().any(axis=1)]

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,Unknown
339,Gentoo,Biscoe,NaN,NaN,NaN,NaN,Unknown


---

### ▶️ Wie viele vollständige Datensätze gibt es?

In [14]:
df.dropna(inplace=True)

In [15]:
df.shape

(342, 7)

🔍 **Antwort:**  
Es bleiben ca. 342 vollständige Zeilen von 344 übrig – sehr wenig fehlende Werte, wir können gut damit arbeiten.

---

## ❓ Forschungsfrage 2: **Was ist der Durchschnitts-Pinguin – und gibt es ihn überhaupt?**

### ▶️ Deskriptive Statistik

In [17]:
df.describe(include="all")

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
count,342,342,342.000000,342.000000,342.000000,342.000000,342
unique,3,3,NaN,NaN,NaN,NaN,3
top,Adelie,Biscoe,NaN,NaN,NaN,NaN,Male
freq,151,167,NaN,NaN,NaN,NaN,168
mean,NaN,NaN,43.921930,17.151170,200.915205,4201.754386,NaN
std,NaN,NaN,5.459584,1.974793,14.061714,801.954536,NaN
min,NaN,NaN,32.100000,13.100000,172.000000,2700.000000,NaN
25%,NaN,NaN,39.225000,15.600000,190.000000,3550.000000,NaN
50%,NaN,NaN,44.450000,17.300000,197.000000,4050.000000,NaN
75%,NaN,NaN,48.500000,18.700000,213.000000,4750.000000,NaN


🔍 **Antwort:**  
- Der durchschnittliche Pinguin hat ~44mm Schnabellänge, ~200mm Flipperlänge, und wiegt ~4200g.
- Die Standardabweichungen zeigen, dass besonders das Körpergewicht stark streut.

---

### ▶️ Einzelwerte ermitteln:

In [19]:
df["body_mass_g"].mean()

4201.754385964912

In [20]:
df["body_mass_g"].median()

4050.0

In [21]:
df["body_mass_g"].std()

801.9545356980954

---

## ❓ Forschungsfrage 3: **Gibt es messbare Unterschiede zwischen Arten oder Geschlechtern?**

### ▶️ Durchschnittliches Gewicht je Art

In [23]:
df.groupby("species")["body_mass_g"].mean()

species
Adelie       3700.662252
Chinstrap    3733.088235
Gentoo       5076.016260
Name: body_mass_g, dtype: float64

---

### ▶️ Durchschnittswerte je Art und Geschlecht

In [25]:
df.groupby(["species", "sex"])["body_mass_g"].agg(["mean", "std"])

mean         std
species   sex                             
Adelie    Female   3368.835616  269.380102
          Male     4043.493151  346.811553
          Unknown  3540.000000  477.166114
Chinstrap Female   3527.205882  285.333912
          Male     3938.970588  362.137550
Gentoo    Female   4679.741379  281.578294
          Male     5484.836066  313.158596
          Unknown  4587.500000  338.193731

🔍 **Antwort:**  

Männchen sind bei allen Arten etwas schwerer als Weibchen.  
Bei Gentoo ist der Unterschied am stärksten.
Unknown beschreibt vermutlich Female.

---

## ❓ Forschungsfrage 4: **Gibt es Ausreißer – und was sagen sie uns?**

### ▶️ IQR-Methode anwenden

In [27]:
q1 = df["body_mass_g"].quantile(0.25)
q3 = df["body_mass_g"].quantile(0.75)
iqr = q3 - q1

lower_bound = q1 - 1.5 * iqr
upper_bound = q3 + 1.5 * iqr

df_outliers = df[(df["body_mass_g"] < lower_bound) | (df["body_mass_g"] > upper_bound)]
df_outliers

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex


Es gibt insgesamt betrachtet keine Ausreißer. Kann man Ausreißer innerhalb einer Gruppe identifizieren?

In [29]:
# Liste zum Sammeln der Ausreißer-Zeilen
outlier_rows = []

# Gruppieren nach Art und Geschlecht (inkl. "Unknown")
grouped = df.dropna(subset=["body_mass_g"]).groupby(["species", "sex"])

for (species, sex), group in grouped:
    q1 = group["body_mass_g"].quantile(0.25)
    q3 = group["body_mass_g"].quantile(0.75)
    iqr = q3 - q1
    lower = q1 - 1.5 * iqr
    upper = q3 + 1.5 * iqr

    outliers = group[(group["body_mass_g"] < lower) | (group["body_mass_g"] > upper)]
    outlier_rows.append(outliers)

# Alle Ausreißer in einem DataFrame zusammenfassen
df_outliers_grouped = pd.concat(outlier_rows)
df_outliers_grouped

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
190,Chinstrap,Dream,46.9,16.6,192.0,2700.0,Female
189,Chinstrap,Dream,52.0,20.7,210.0,4800.0,Male
246,Gentoo,Biscoe,44.5,14.3,216.0,4100.0,Unknown


## ❓ Forschungsfrage 5: **Welche Merkmale hängen miteinander zusammen?**

### ▶️ Korrelation berechnen (numerisch)
Diese kann zwischen -1 und 1 liegen, wobei ein hoher bzw. niedriger Wert auf einen Zusammenhang hindeutet, und ein Wert nahe 0 auf keinen.
Dies sind aber keine Kausalitäten!

In [31]:
df.corr(numeric_only=True)

,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g
bill_length_mm,1.000000,-0.235053,0.656181,0.595110
bill_depth_mm,-0.235053,1.000000,-0.583851,-0.471916
flipper_length_mm,0.656181,-0.583851,1.000000,0.871202
body_mass_g,0.595110,-0.471916,0.871202,1.000000


🔍 **Antwort:**  
- **Flipperlänge und Körpergewicht** haben eine starke positive Korrelation (~0.87) → größere Flossen = schwerere Pinguine.
- Schnabellänge korreliert ebenfalls leicht mit Körpergewicht.

---

## ✅ Fazit

| Gelernt | Beispiel |
|--------|----------|
| Überblick verschaffen | `head()`, `info()`, `shape` |
| Fehlende Werte finden | `isnull().sum()`, `dropna()` |
| Deskriptive Statistik | `describe()`, `mean()`, `std()` |
| Gruppen vergleichen | `groupby()`, `agg()` |
| Ausreißer erkennen | IQR, `quantile()`, Filterung |
| Korrelationen | `.corr()` |

---